python/lsst/daf/butler/core/schema.py

```diff
@@ -83,317 +116,341 @@ def process_result_value(self, value, dialect):
-            return True
-        description = self.config["tables"][name]
-        return "sql" in description and not description.get("materialize", False)
+        dtype = VALID_COLUMN_TYPES.get(config["type"])
+        if dtype is None:
+            raise SchemaValidationError(f"Invalid field type string: '{config['type']}'.")
+        if not config["name"].islower():
+            raise SchemaValidationError(f"Column name '{config['name']}' is not all lowercase.")
+        self = cls(name=config["name"], dtype=dtype, **kwds)
+        self.length = config.get("length", self.length)
+        self.nbytes = config.get("nbytes", self.nbytes)
+        if self.length is not None and self.nbytes is not None:
+            raise SchemaValidationError(f"Both length and nbytes provided for field '{self.name}'.")
+        self.primaryKey = config.get("primaryKey", self.primaryKey)
+        self.autoincrement = config.get("autoinc", self.autoincrement)
+        self.nullable = config.get("nullable", False if self.primaryKey else self.nullable)
+        self.doc = stripIfNotNone(config.get("doc", None))
+        return self
+
+    def getSizedColumnType(self):
+        """Return a sized version of the column type, utilizing either (or
+        neither) of ``self.length`` and ``self.nbytes``.
+        """
+        if self.length is not None:
+            return self.dtype(length=self.length)
+        if self.nbytes is not None:
+            return self.dtype(nbytes=self.nbytes)
+        return self.dtype
 
-    def isIncluded(self, name):
-        """Return True if the named table or view should be included in this
-        schema.
+    def toSqlAlchemy(self, tableName: str, schema: Schema) -> Column:
+        """Construct a SQLAlchemy `Column` object from this specification.
 
         Parameters
         ----------
-        name : `str`
-            Name of a table or view.  Does not need to have been added.
-
-        Returns
-        -------
-        included : `bool`
-            Whether the table or view should be included in the schema.
+        tableName : `str`
+            Name of the logical table to which this column belongs.
+        schema : `Schema`
+            Object represening the full schema.  May be modified in-place.
         """
-        if name in self.tables:
-            return True
-        description = self.config["tables"].get(name, None)
-        if description is None:
-            return False
-        return True
+        args = [self.name, self.getSizedColumnType()]
+        if self.autoincrement:
+            # Generate a sequence to use for auto incrementing for databases
+            # that do not support it natively.  This will be ignored by
+            # sqlalchemy for databases that do support it.
+            args.append(Sequence(f"{tableName}_{self.name}_seq", metadata=schema.metadata))
+        return Column(*args, nullable=self.nullable, primary_key=self.primaryKey, comment=self.doc)
+
+
+@dataclass
```

Comment:
Well, tests pass if I do mark it `frozen`, at least.  I'm not certain it's a good idea here, because I *have* found use cases for mutating `TableSpec` and `FieldSpec` (usually some kind of "make a modified copy" operation), and because the lifetimes and sharing of of these objects is generally limited.